xarray/tests/test_dataset.py

```diff
@@ -4947,15 +4947,16 @@ def test_reduce_cumsum_test_dims(self, reduct, expected, func):
     def test_reduce_non_numeric(self):
         data1 = create_test_data(seed=44)
         data2 = create_test_data(seed=44)
-        add_vars = {"var4": ["dim1", "dim2"]}
+        add_vars = {"var4": ["dim1", "dim2"], "var5": ["dim1"]}
         for v, dims in sorted(add_vars.items()):
             size = tuple(data1.dims[d] for d in dims)
             data = np.random.randint(0, 100, size=size).astype(np.str_)
             data1[v] = (dims, data, {"foo": "variable"})
 
-        assert "var4" not in data1.mean()
+        assert all(var not in data1.mean() for var in add_vars)
```

Comment:
I wonder if it's easier to keep the explicit checks, especially since we have a similar check 3 lines later?
```suggestion
        assert "var4" not in data1.mean() and "var5" not in data1.mean()
```